thunder/executors/pythonex.py

```diff
@@ -49,7 +49,11 @@ def _check_tensor_shape_and_metadata_impl(
 ) -> None:
     assert isinstance(t, torch.Tensor), f"expected Tensor, got {type(t).__name__}"
     assert (
-        tuple(t.shape) == shape and str(t.device) == device and t.dtype == dtype and t.requires_grad == requires_grad
+        len(t.shape) == len(shape)
+        and all(map(lambda x, ref: ref == -1 or x == ref, t.shape, shape))
```

Comment:
Why does the prim not take the actual number proxies for shape instead of converting them to -1?

Reply:
This is just for simplification.
Say if we do take the actual number proxies here, what are we going to use it for? I assume it's serving the same purpose, it'll just be a wild card and we'll skip any checks on the entry.

 I don't hold a strong opinion on this. Do you think keeping a NumberProxy is easier to read, vs having a random `-1` holding a special meaning?

Reply:
I don't think it matters much today. I have a preference for passing the actual number proxies, if possible, as the number proxies may be augmented with data in the future that the prim could inspect

Reply:
Sounds good. Let me try to add it there as a placeholder